contrib/python/api/skydive/websocket/client.py

```diff
@@ -208,13 +219,32 @@ def login(self, host_spec, username, password):
         :type password: string
         :return: True on successful authentication, False otherwise
         """
-        res = requests.post(
-            'http://{0}/login'.format(host_spec),
-            data={
-                'username': username,
-                'password': password,
-            },
-        )
+
+        if host_spec:
```

Comment:
I would rather do:

```
endpoint = self.endpoint
if host_spec:
   endpoint = host_spec

if not username:
   username = self.username

if not password:
   password = self.password
```
so that only one path, using the "else" one which handles ssl and port